api/src/shop/transactions.py

```diff
@@ -342,11 +342,14 @@ def payment_success(transaction: Transaction) -> None:
 
 def process_cart(member_id: int, cart: List[CartItem]) -> Tuple[Decimal, List[TransactionContent]]:
     contents = []
+    labaccess_in_cart = False
+    base_membership_in_cart = False
 
     member = db_session.get(Member, member_id)
     if member is None:
         raise NotFound(message=f"Could not find member with id {member_id}.")
     price_level = get_price_level_for_member(member)
+    member_has_base_membership = Span.MEMBERSHIP in [span.type for span in member.spans]
```

Comment:
Does this check really work if the base membership span has ended? Like a person buys base membership one year, then the next year when the base membership has ended, then, would they be able to buy labaccess?